src/main/java/seedu/address/storage/JsonAdaptedInternship.java

```diff
@@ -72,7 +74,55 @@ public Internship toModelType() throws IllegalValueException {
             tags.add(tag.toModelType());
         }
 
-        return new Internship(new Name(name), role, new Time(from), new Time(to), description, Set.copyOf(tags), id);
+        if (name == null) {
+            throw new IllegalValueException(String.format(MISSING_FIELD_MESSAGE_FORMAT, Name.class.getSimpleName()));
+        }
+        if (!Name.isValidName(name)) {
+            throw new IllegalValueException(Name.MESSAGE_CONSTRAINTS);
+        }
+        final Name modelName = new Name(name);
+
+        if (role == null) {
+            throw new IllegalValueException(String.format(MISSING_FIELD_MESSAGE_FORMAT, "role"));
+        }
+
+        if (from == null) {
+            throw new IllegalValueException(String.format(MISSING_FIELD_MESSAGE_FORMAT,
+                    Time.class.getSimpleName()));
+        }
+        if (!Time.isValidTime(from)) {
+            throw new IllegalValueException(Time.MESSAGE_CONSTRAINTS);
+        }
+        final Time modelFrom = new Time(from);
+
+        if (to == null) {
+            throw new IllegalValueException(String.format(MISSING_FIELD_MESSAGE_FORMAT,
+                    Time.class.getSimpleName()));
+        }
+        if (!Time.isValidTime(to)) {
+            throw new IllegalValueException(Time.MESSAGE_CONSTRAINTS);
+        }
+        final Time modelTo = new Time(to);
+
+        // Enforces that to does not precede from
+        if (modelTo.compareTo(modelFrom) < 0) {
+            throw new IllegalValueException("The \"to\" field must not precede the \"from\" field.");
```

Comment:
Just want to comment that this String is already defined in a few places, can refactor later

Reply:
Same comment here. A lot of String literals all over the place that could be abstracted.